kvision-tools/kvision-gradle-plugin/src/main/kotlin/io/kvision/gradle/KVisionPlugin.kt

```diff
@@ -0,0 +1,369 @@
+     * Initialise the [KVisionExtension] on a [Project].
+     *
+     * Additionally, set config that requires a [Project] instance.
+     */
+    private fun Project.createKVisionExtension(): KVisionExtension {
+        return extensions.create("kvision", KVisionExtension::class).apply {
+            nodeBinaryPath.convention(nodeJsBinaryProvider())
+
+            kotlinJsStoreDirectory.convention(project.layout.projectDirectory.dir(".kotlin-js-store"))
+        }
+    }
+
+
+    /**
+     * Helper class that provides both a Gradle [Project] and [KVisionExtension].
+     *
+     * Useful for extension functions that configure a project.
+     */
+    private data class KVPluginContext(
+        private val project: Project,
+        val kvExtension: KVisionExtension,
+    ) : Project by project
+
+
+    /** Configure a Kotlin JS project */
+    private fun KVPluginContext.configureJsProject() {
+
+        val kotlinJsExtension = extensions.getByType<KotlinJsProjectExtension>()
+
+        registerGeneratePotFileTask {
+            dependsOn(tasks.compileKotlinJs)
+
+            inputs.files(kotlinJsExtension.sourceSets.main.get().kotlin.files)
+
+            potFile.set(
+                layout.projectDirectory.file(
+                    "src/main/resources/i18n/messages.pot"
+                )
+            )
+        }
+
+        registerConvertPoToJsonTask {
+            dependsOn(tasks.compileKotlinJs)
+
+            sourceDirectory.set(
+                layout.dir(tasks.processResources.map { it.destinationDir })
+            )
+        }
+
+        registerZipTask()
+
+        afterEvaluate {
+            tasks.processResources.configure {
+                exclude("**/*.pot")
+                dependsOn(tasks.compileKotlinJs)
+            }
+        }
+    }
+
+
+    /** Configure a Kotlin Multiplatform project */
+    private fun KVPluginContext.configureMppProject() {
+
+        val kotlinMppExtension = extensions.getByType<KotlinMultiplatformExtension>()
+
+        registerGeneratePotFileTask {
+            dependsOn(tasks.compileKotlinFrontend)
+            inputs.files(kotlinMppExtension.sourceSets.frontendMain.map { it.kotlin.files })
+            potFile.set(
+                layout.projectDirectory.file(
+                    "src/frontendMain/resources/i18n/messages.pot"
+                )
+            )
+        }
+
+        registerConvertPoToJsonTask {
+            dependsOn(tasks.compileKotlinFrontend)
+            sourceDirectory.set(
+                layout.dir(tasks.frontendProcessResources.map { it.destinationDir })
+            )
+        }
+
+        registerWorkerBundleTask()
+
+        tasks.getByName("compileKotlinFrontend") {
+            dependsOn("compileCommonMainKotlinMetadata")
+        }
+        tasks.create("generateKVisionSources") {
+            group = "KVision"
+            description = "Generates KVision sources for fullstack interfaces"
+            dependsOn("compileCommonMainKotlinMetadata")
+        }
```

Comment:
This task is copied directly: https://github.com/rjaros/kvision/blob/90dcbbdfe60e17cd5a1cda7de577bf54b8716549/kvision-tools/io.kvision.gradle.plugin/src/main/kotlin/io/kvision/gradle/KVisionGradleSubplugin.kt#L147-L151

What's the purpose of this task? It doesn't have any inputs or outputs, and doesn't seem to be used.

Reply:
It was requested by one of the users to create a dedicated task for KVision sources generation. The compiler plugin is attached to the compilation process so it's not clear what task should be called when e.g. listing gradle tasks.